libsolidity/codegen/ArrayUtils.h

```diff
@@ -77,9 +73,10 @@ class ArrayUtils
 	/// Stack post:
 	void popStorageArrayElement(ArrayType const& _type) const;
 	/// Appends a loop that clears a sequence of storage slots of the given type (excluding end).
+	/// @param _canOverflow whether the storage is treated as circular when clearing.
 	/// Stack pre: end_ref start_ref
 	/// Stack post: end_ref
-	void clearStorageLoop(Type const* _type) const;
+	void clearStorageLoop(Type const* _type, bool _canOverflow) const;
```

Comment:
I'll have more concrete suggestions for the bug description and naming later, but one thing I'd like to point out already is that "overflow" sounds a bit ambiguous to me in the context of this bug. It could mean many things. I think that "wrap around" would generally be clearer.